### PR TITLE
fix some unit tests [dev3x]

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

File: src/Microsoft.Identity.Client/PublicClientApplication.WithBuilders.cs [hunk 1]
```diff
@@ -129,24 +129,7 @@ internal PublicClientApplication(ApplicationConfiguration configuration)
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
-        internal async Task<AuthenticationResult> ExecuteAsync(
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/PublicClientApplication.WithBuilders.cs]: >internal [](start = 8, length = 8)

did the silentRequest get moved somewhere else?

File: tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs [hunk 2]
```diff
@@ -82,7 +82,7 @@ public void TestInitialize()
         internal void SetupMocks(MockHttpManager httpManager)
         {
             httpManager.AddInstanceDiscoveryMockHandler();
-            httpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.AuthorityHomeTenant);
+            httpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.AuthorityCommonTenant);
```
[the remark this answers is not in the document]
Comment (MarkZuber) [reply tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs]: I added the URL that's being called to the mock handler across the board ehre.  My initial guess as to the behavior was wrong, these are hitting the common tenant in reality.  If this test code should be hitting home tenant instead in these cases then something's broken.

---
In reply to: [248113861](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/788#discussion_r248113861) [](ancestors = 248113861)

File: src/Microsoft.Identity.Client/ApiConfig/IAcquireTokenOnBehalfOfParameters.cs [hunk 3]
```diff
@@ -37,6 +37,6 @@ internal interface IAcquireTokenOnBehalfOfParameters : IAcquireTokenCommonParame
 
         /// <summary>
         /// </summary>
-        bool WithOnBehalfOfCertificate { get; }
+        bool SendX5C { get; }
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/ApiConfig/IAcquireTokenOnBehalfOfParameters.cs]: >SendX5C [](start = 13, length = 7)

We should ensure the explanation of the usage of this one is clear. It's used for the subjectname + issuer feature allowing evo to roll certs with-out having to have both certs in the system when rolling

File: tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (henrik-me) [reply tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs]: typically the home tenant is used for storing the token, however it's very likely that the actual authority used would be using the common end-point for the test cases.

---
In reply to: [248115639](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/788#discussion_r248115639) [](ancestors = 248115639,248113861)

[the remark this answers is not in the document]
Comment (henrik-me) [reply tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs]: but it should certainly be validated what the intent of the test is (seems like this pattern is in many places)

---
In reply to: [248134636](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/788#discussion_r248134636) [](ancestors = 248134636,248115639,248113861)

**Pull request merged**